internal/api/router_test.go

```diff
@@ -1,33 +1,34 @@
 package api
 
 import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"ship-it/internal/api/models"
 	"testing"
 
-	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics/discard"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestRoutes(t *testing.T) {
-	expectedRoutes := map[string]interface{}{
-		"/*":      nil,
-		"/health": nil,
-		"/api/*": map[string]interface{}{
-			"/releases": nil,
-		},
-	}
+type nopService struct{}
 
-	handler := New(nil, nil)
+func (m *nopService) ListReleases(ctx context.Context) ([]models.Release, error) {
+	return nil, nil
+}
 
-	if routes, ok := handler.(chi.Routes); ok {
-		assertRoutes(t, expectedRoutes, routes)
+func TestRoutes(t *testing.T) {
+	requests := []*http.Request{
```

Comment:
This treats the router as a black box `http.Handler`, but there's some things I dislike about this test compared to what I had previously:

* I need to provide concrete implementations of the router's `Service` and `Histogram` dependencies. These are only relevant to the controller business logic which is thoroughly tested elsewhere. For the router tests I only care about the structure of the routing table, not the handlers.

* it tests less than the previous approach. The `/` route is served by a filesystem handler, so `/index.html` matches but also `/favicon.ico` and dynamic assets like `/static/main.5ee0bde0.chunk.css`. I can't test that requests for all these dynamic paths are handled correctly.

* same issues as originally, what realistic code change would break this test?